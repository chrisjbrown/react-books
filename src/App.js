import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import styled from "styled-components";
import { Provider } from "react-redux";

import Bookshelf from "./pages/BookShelf";
import Book from "./pages/Book";
import store from "./store";

const Layout = styled.div`
  max-width: 880px;
  margin: 0 auto;
`;

function App() {
  return (
    <Router>
      <Layout>
        <Provider store={store}>
          <Switch>
            <Route exact path="/">
              <Bookshelf />
            </Route>
            <Route path="/book/:id">
              <Book />
            </Route>
          </Switch>
        </Provider>
      </Layout>
    </Router>
  );
}

export default App;
