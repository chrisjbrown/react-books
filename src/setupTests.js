import "@testing-library/jest-dom";
import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import "jest-styled-components";

export function renderWithRouter(
  ui,
  { route = "/", initialState = [], ...renderOptions } = {}
) {
  window.history.pushState({}, "Test page", route);
  const mockStore = configureStore([]);
  const store = mockStore(initialState);
  function Wrapper({ children }) {
    return (
      <BrowserRouter>
        <Provider store={store}>{children}</Provider>
      </BrowserRouter>
    );
  }

  return { ...render(ui, { wrapper: Wrapper, ...renderOptions }), store };
}
