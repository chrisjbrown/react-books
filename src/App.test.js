import React from "react";
import { renderWithRouter } from "./setupTests";

import App from "./App";

it("should have default route", () => {
  const { container } = renderWithRouter(<App />);

  expect(container.firstChild).toMatchSnapshot();
});

it("should have book route", () => {
  const { container } = renderWithRouter(<App />, {
    route: "/book/123",
  });
  expect(container.firstChild).toMatchSnapshot();
});
