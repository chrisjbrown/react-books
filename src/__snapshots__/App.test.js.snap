// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have book route 1`] = `
.c1 {
  text-align: center;
}

.c2 {
  text-align: center;
  color: red;
}

.c0 {
  max-width: 880px;
  margin: 0 auto;
}

<div
  class="c0"
>
  <div>
    <h1
      class="c1"
    >
      Book
    </h1>
    <a
      href="/"
    >
      <svg
        fill="currentColor"
        height="2em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 576 512"
        width="2em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
        />
      </svg>
    </a>
    <h2>
      Book details
    </h2>
    <div
      class="c2"
    >
      Book not found :(
    </div>
  </div>
</div>
`;

exports[`should have default route 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border: 0;
  padding: 10px 20px;
  cursor: pointer;
  outline: 0;
}

.c3:active {
  background-color: lightblue;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border: 1px solid lightblue;
  padding: 10px 20px;
  cursor: pointer;
  outline: 0;
}

.c4:active {
  background-color: lightblue;
}

.c1 {
  text-align: center;
}

.c5 {
  display: grid;
  grid-template-columns: repeat(auto-fit,160px);
  -webkit-column-gap: 20px;
  column-gap: 20px;
  row-gap: 20px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 10px;
  border: 1px solid black;
  margin-bottom: 20px;
}

.c0 {
  max-width: 880px;
  margin: 0 auto;
}

<div
  class="c0"
>
  <div>
    <h1
      class="c1"
    >
      Bookshelf
    </h1>
    <div
      class="c2"
    >
      <button
        class="c3"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
          />
        </svg>
        Sort
      </button>
      <button
        class="c4"
      >
        + 
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            add-book
          </title>
          <path
            d="M448 360V24c0-13.3-10.7-24-24-24H96C43 0 0 43 0 96v320c0 53 43 96 96 96h328c13.3 0 24-10.7 24-24v-16c0-7.5-3.5-14.3-8.9-18.7-4.2-15.4-4.2-59.3 0-74.7 5.4-4.3 8.9-11.1 8.9-18.6zM128 134c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm0 64c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm253.4 250H96c-17.7 0-32-14.3-32-32 0-17.6 14.4-32 32-32h285.4c-1.9 17.1-1.9 46.9 0 64z"
          />
        </svg>
      </button>
      <button
        class="c3"
      >
        <span>
          List/
          <strong>
            Grid
          </strong>
        </span>
      </button>
    </div>
    <div
      class="c5"
    />
  </div>
</div>
`;
